sdk/src/main/scala/com/horizen/account/state/AccountState.scala

```diff
@@ -148,7 +148,7 @@ class AccountState(
 
             // get a receipt obj with non consensus data (logs updated too)
             val fullReceipt =
-              EthereumReceipt(consensusDataReceipt, txHash, txIndex, blockHash, blockNumber, txGasUsed, contractAddress)
+              EthereumReceipt(consensusDataReceipt, txHash, txIndex, blockHash, blockNumber, txGasUsed, Option(contractAddress))
```

Comment:
A few line above, I expect `val contractAddress: Option[Byte[Array]]`.
And in case it is not a smart contract declaration -> set it to `None` instead of `new Array[Byte](0)`